### PR TITLE
Rule to check RDS snapshot is not shared publicly.

### PR DESCRIPTION
**Pull request by JKeeter**
I confirm these files are made available under CC0 1.0 Universal (https://creativecommons.org/publicdomain/zero/1.0/legalcode)

### CODE REVIEW

File: python/rds_public_snapshots.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+log = logging.getLogger()
+log.setLevel(logging.INFO)
+
+def evaluate_compliance(configuration_item):
+    public = False
+    for v in configuration_item["supplementaryConfiguration"]["DBSnapshotAttributes"][0]["attributeValues"]:
+        if v == "all":
+            public = True
+            continue
```
Comment (jongogogo) [inline python/rds_public_snapshots.py]: Better to do a
``break``

File: python/rds_public_snapshots.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    if public:
+        return {
+            "compliance_type" : "NON_COMPLIANT",
+            "annotation" : 'Snapshot shared publicly'
```
Comment (jongogogo) [inline python/rds_public_snapshots.py]: Full sentence, please

File: python/rds_public_snapshots.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        }
+    else:
+        return {
+            "compliance_type": "COMPLIANT",
+            "annotation": 'Snapshot not shared publicly'
```
Comment (jongogogo) [inline python/rds_public_snapshots.py]: Full sentence, please

File: python/rds_public_snapshots.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        }
+    
+
+def lambda_handler(event, context):
+    log.debug('Event %s', event) 
+    invoking_event      = json.loads(event['invokingEvent'])
```
Comment (jongogogo) [inline python/rds_public_snapshots.py]: The code does not handle a deleted resource.
Have you considered to use the Rule Development Kit, which does that for you? https://github.com/awslabs/aws-config-rdk

**Comment by JKeeter** [comment]
Hi - requested changes have been added.

File: python/rds_public_snapshots.py [hunk 5]
```diff
@@ -20,28 +20,36 @@ def evaluate_compliance(configuration_item):
...
+    return (status == 'OK' or status == 'ResourceDiscovered') and eventLeftScope == False
+
 
 def lambda_handler(event, context):
     log.debug('Event %s', event) 
     invoking_event      = json.loads(event['invokingEvent'])
     configuration_item  = invoking_event["configurationItem"]
-    evaluation          = evaluate_compliance(configuration_item)
+    if is_applicable(configuration_item,event):
+      evaluation        = evaluate_compliance(configuration_item)
```
Comment (jongogogo) [inline python/rds_public_snapshots.py]: Increment and style not consistent with the previous code.

File: python/rds_public_snapshots.py [hunk 6]
```diff
@@ -20,28 +20,36 @@ def evaluate_compliance(configuration_item):
...
+    else:
+      evaluation	= {"compliance_type":"NOT_APPLICABLE","annotation":"The RDS Snapshot has been deleted."}
```
Comment (jongogogo) [inline python/rds_public_snapshots.py]: same

File: python/rds_public_snapshots.py [hunk 7]
```diff
@@ -20,28 +20,36 @@ def evaluate_compliance(configuration_item):
...
     config              = boto3.client('config')
```
Comment (jongogogo) [inline python/rds_public_snapshots.py]: same

**Pull request merged**